### PR TITLE
fix(cnsl): hide new auth button if no routerlink

### LINKED ISSUE #6290: [Bug]: The "New" button from Authorizations is shown even if the user is not in charge of managing them
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Version

v2.31.1

### Describe the problem caused by this bug

The User User is just a simple user, but the UI shows the New button
However, this button does nothing.
It probably makes sense to hide the button.

### To reproduce

Steps to reproduce:
1. Be a simple user with no special permissions
2. Open Zitadel UI
3. Go to Authorizations
4. See the button which does nothing

### Screenshots

Screenshot 2023-07-28 at 15 50 43

### Expected behavior

- Either no button could be observed
- Or button is disabled

### Operating System

Mac
macOS
Chrome

### Relevant Configuration

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by doncicuto**
In #6290 Vladimir Logachev detected a bug where the New auth button was shown but clicking on it did nothing (as expected) for a simple user.

It seems that user.grant.write is not the only condition to hide that button in these situations, and I have checked that the button should be hidden if routerLink has no path set. A new showAddButton property is added.

In this screenshot the simple user no longer see the New button:

image

Here Zitadel Admin can add new auths:

image

### Note for reviewer

E2E every test run successfully with the exception of events/event.cy.ts which in my opinion has nothing to do with this code

Closes #6290

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: console/src/app/modules/user-grants/user-grants.component.html [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
   <a
     actions
-    *ngIf="disableWrite === false && selection.hasValue() === false"
+    *ngIf="disableWrite === false && selection.hasValue() === false && showAddButton"
```
Comment (peintnermax) [inline console/src/app/modules/user-grants/user-grants.component.html:33]:
```suggestion
    *ngIf="disableWrite === false && selection.hasValue() === false && routerLink"
```

File: console/src/app/modules/user-grants/user-grants.component.ts [hunk 2]
```diff
@@ -61,6 +61,7 @@ export class UserGrantsComponent implements OnInit, AfterViewInit {
   public loadedId: string = '';
   public loadedProjectId: string = '';
   public grantToEdit: string = '';
+  public showAddButton: boolean = false;
```
Comment (peintnermax) [inline console/src/app/modules/user-grants/user-grants.component.ts:64]:
```suggestion
```

File: console/src/app/modules/user-grants/user-grants.component.ts [hunk 3]
```diff
@@ -93,25 +94,30 @@ export class UserGrantsComponent implements OnInit, AfterViewInit {
       case UserGrantContext.OWNED_PROJECT:
         if (this.projectId) {
           this.routerLink = ['/grant-create', 'project', this.projectId];
+          this.showAddButton = true;
```
Comment (peintnermax) [inline console/src/app/modules/user-grants/user-grants.component.ts:97]:
```suggestion
```

File: console/src/app/modules/user-grants/user-grants.component.ts [hunk 4]
```diff
@@ -93,25 +94,30 @@ export class UserGrantsComponent implements OnInit, AfterViewInit {
...
         }
         break;
       case UserGrantContext.GRANTED_PROJECT:
         if (this.grantId) {
           this.routerLink = ['/grant-create', 'project', this.projectId, 'grant', this.grantId];
+          this.showAddButton = true;
```
Comment (peintnermax) [inline console/src/app/modules/user-grants/user-grants.component.ts:103]:
```suggestion
```

File: console/src/app/modules/user-grants/user-grants.component.ts [hunk 5]
```diff
@@ -93,25 +94,30 @@ export class UserGrantsComponent implements OnInit, AfterViewInit {
...
         }
         break;
       case UserGrantContext.USER:
         if (this.userId) {
           this.routerLink = ['/grant-create', 'user', this.userId];
+          this.showAddButton = true;
```
Comment (peintnermax) [inline console/src/app/modules/user-grants/user-grants.component.ts:109]:
```suggestion
```

File: console/src/app/modules/user-grants/user-grants.component.ts [hunk 6]
```diff
@@ -93,25 +94,30 @@ export class UserGrantsComponent implements OnInit, AfterViewInit {
...
         }
         break;
       case UserGrantContext.AUTHUSER:
         if (this.grantId) {
           this.routerLink = ['/grant-create', 'user', this.userId];
+          this.showAddButton = true;
```
Comment (peintnermax) [inline console/src/app/modules/user-grants/user-grants.component.ts:115]:
```suggestion
```

File: console/src/app/modules/user-grants/user-grants.component.ts [hunk 7]
```diff
@@ -93,25 +94,30 @@ export class UserGrantsComponent implements OnInit, AfterViewInit {
...
         }
         break;
       case UserGrantContext.NONE:
         this.routerLink = ['/grant-create'];
+        this.showAddButton = true;
```
Comment (peintnermax) [inline console/src/app/modules/user-grants/user-grants.component.ts:120]:
```suggestion
```

File: console/src/app/modules/user-grants/user-grants.component.html (hunk 1 above)
Comment (peintnermax) [reply console/src/app/modules/user-grants/user-grants.component.html:33]: lets use the `routerLink` variable itself to check if the button should display or not

**Review by peintnermax — CHANGES_REQUESTED** [review]
Thx for your contribution! 👌I'd use the routerLink variable itself to check if it is set so we keep the loc to a minimum

File: console/src/app/modules/user-grants/user-grants.component.ts (hunk 2 above)
Comment (peintnermax) [reply console/src/app/modules/user-grants/user-grants.component.ts:64]: we have to initialize the routerLink with undefined in order to check it in the html.
```
public routerLink: any = undefined;
```

**Comment by doncicuto** [comment]
You're right @peintnermax, your solution is prettier and cleaner

Thanks a lot... bug fixed 😄

**Comment by peintnermax** [comment]
> You're right @peintnermax, your solution is prettier and cleaner
> 
> Thanks a lot... bug fixed 😄

Thx for your fast reaction 👍👍

**Review by peintnermax — APPROVED** [review]

**Pull request merged**